account/identity.go

```diff
@@ -140,10 +142,18 @@ func (m *GormIdentityRepository) Create(ctx context.Context, model *Identity) er
 	}
 	err := m.db.Create(model).Error
 	if err != nil {
-		goa.LogError(ctx, "error adding Identity", "error", err.Error())
+		log.LogError(ctx, map[string]interface{}{
+			"identity": model,
+			"err":      err.Error(),
+		}, "Unable to create the identity")
 		return errors.WithStack(err)
```

Comment:
@kwk I agree with your example. But the example differs from what is happening in our code. Another demonstration can be illustrated here:

```
// returns BadParameterError, ConversionError or InternalError
func (r *GormWorkItemTypeRepository) Create(ctx context.Context, extendedTypeName *string, name string, fields map[string]app.FieldDefinition) (*app.WorkItemType, error) {
	err := er.New("Oh my god what an error")
	log.Error(ctx, map[string]interface{}{
			"typeName": name,
			"err":        err,
	}, "unable to create the wit")

	return nil, roaming.Wrap(err, "wrapping ..unable to create the wit")
}
```
client output:
```
./bin/alm-cli .... --payload ... -H localhost:8080 --pp
2017/02/17 01:33:34 [INFO] started id=1W8Y8m8A POST=http://localhost:8080/api/workitemtypes
2017/02/17 01:33:34 [INFO] completed id=1W8Y8m8A status=500 time=34.903961ms
error: 500: {"errors":[{"code":"unknown_error","detail":"Oh my god what an error","status":"500","title":"Unknown error"}]}
```
Server logs:
```
INFO[2017-02-17 01:33:34] payload                                       raw="{\"fields\":{\"system.owner\":{\"required\":true,\"type\":{\"kind\":\"user\"}},\"system.state\":{\"required\":false,\"type\":{\"kind\":\"string\"}}},\"name\":\"Epic\"}" req_id=JPFOYPgcfz-3
ERRO[2017-02-17 01:33:34] unable to create the wit                      err="Oh my god what an error" pid=2847 requestID=JPFOYPgcfz-3 typeName=Epic
DEBU[2017-02-17 01:33:34] Commit the transaction!                       pkg=application
INFO[2017-02-17 01:33:34] completed                                     action=Create bytes=112 ctrl=WorkitemtypeController req_id=JPFOYPgcfz-3 status=500 time=6.145473ms
```
This shows how adding the logger here adds value for the server while the `errors.Wrap....` adds it at the client. Both logs are actionable. I hope this helps to end it up this discussion :). We can always iterate over it if we see duplicate info, etc...

On the other hand, the article states that you should log actionable information. If you log the request of an user even if we didn't find any record. It is important to offer: 1) better user experience cause I can see what happened if he asks. 2) I can trace the logs to see the type of errors and which information the users tried to get.. These two are in my opinion actionable information.

As you certainly described with your example, these error messages apply to the same issue so duplicate the error msgs.